src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/LanguageServices/AbstractHostObject.cs

```diff
@@ -0,0 +1,122 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.TextManager.Interop;
+using IOLEServiceProvider = Microsoft.VisualStudio.OLE.Interop.IServiceProvider;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.LanguageServices
+{
+    /// <summary>
+    /// Base type for language service host object for cross targeting projects.
+    /// </summary>
+    internal abstract class AbstractHostObject : IVsHierarchy, IVsContainedLanguageProjectNameProvider
+    {
+        protected abstract IVsHierarchy InnerHierarchy { get; }
```

Comment:
I don't know the answer to this. I am mimicing what xproj did by handling all the properties related to shared projects and delegating the rest to inner hierarchy.